RLTest/__main__.py

```diff
@@ -345,57 +362,25 @@ def execute(self):
             raw_input('press any button to stop')
             env.stop()
             return
-        if self.args.tests_file:
-            self._loadFileTests(self.args.tests_file)
-        else:
-            self._loadTests()
         done = 0
         startTime = time.time()
         if self.args.interactive_debugger and len(self.tests) != 1:
             print Colors.Bred('only one test can be run on interactive-debugger use --test-name')
             sys.exit(1)
-        while self.tests:
+
+        for test in self.loader:
             with self.envScopeGuard():
-                test = self.tests.pop(0)
-                if inspect.isclass(test):
-
-                    # checking if there are tests to run
-                    methodsToTest = []
-                    for m in dir(test):
-                        if self.args.test_name is not None:
-                            if self.args.test_name == m:
-                                methodsToTest.append(m)
-                        elif m.startswith('test') or m.startswith('Test'):
-                            methodsToTest.append(m)
-
-                    if len(methodsToTest) == 0:
-                        continue
+                if test.is_class:
                     try:
-                        testObj = test()
+                        obj = test.create_instance()
```

Comment:
So if exception rises during class creation then we will crash and exit, I prefer to print the exception and continue.

Reply:
Good point. I will fix